cmd/dra-example-controller/gpu.go

```diff
@@ -59,6 +59,35 @@ func (g *gpudriver) Allocate(crd *nascrd.NodeAllocationState, claim *resourcev1.
 	return onSuccess, nil
 }
 
+func (g *gpudriver) AllocateImmediately(crd *nascrd.NodeAllocationState, ca *controller.ClaimAllocation) error {
+	claimUID := string(ca.Claim.UID)
+
+	cas := []*controller.ClaimAllocation{ca}
+	allocated := g.allocate(crd, cas)
+
+	claimParams := ca.ClaimParameters.(*gpucrd.GpuClaimParametersSpec)
+	if claimParams.Count != len(allocated[claimUID]) {
+		return fmt.Errorf("not enough resources")
+	}
+
+	var devices []nascrd.RequestedGpu
+	for _, gpu := range allocated[claimUID] {
+		device := nascrd.RequestedGpu{
+			UUID: gpu,
+		}
+		devices = append(devices, device)
+	}
+
+	requestedDevices := nascrd.RequestedDevices{
+		Gpu: &nascrd.RequestedGpus{
+			Devices: devices,
+		},
+	}
```

Comment:
Can we factor this out to a function so that it can be reused in unsuitable nodes and here?

 Ideally this function and unsuitable nodes would be almost line-for-line identical (and very short) except for putting the request into `PendingClaimRequests`.

Reply:
Done.